src/main/cms/cms_menu_imu.c

```diff
@@ -227,6 +228,110 @@ static CMS_Menu cmsx_menuPid = {
+
+    return 0;
+}
+
+static long cmsx_slidersOnExit(const OSD_Entry *self)
+{
+    UNUSED(self);
+
+    pidProfile_t *pidProfile = currentPidProfile;
+
+    pidProfile->slider_pids_mode = cmsx_slider_pids_mode;
+    pidProfile->slider_master_multiplier = cmsx_slider_master_multiplier;
+    pidProfile->slider_roll_pitch_ratio = cmsx_slider_roll_pitch_ratio;
+    pidProfile->slider_i_gain = cmsx_slider_i_gain;
+    pidProfile->slider_pd_ratio = cmsx_slider_pd_ratio;
+    pidProfile->slider_pd_gain = cmsx_slider_pd_gain;
+    pidProfile->slider_dmin_ratio = cmsx_slider_dmin_ratio;
+    pidProfile->slider_ff_gain = cmsx_slider_ff_gain;
+
+    pidProfile->slider_dterm_filter = cmsx_slider_dterm_filter;
+    pidProfile->slider_dterm_filter_multiplier = cmsx_slider_dterm_filter_multiplier;
+    gyroConfigMutable()->slider_gyro_filter = cmsx_slider_gyro_filter;
+    gyroConfigMutable()->slider_gyro_filter_multiplier = cmsx_slider_gyro_filter_multiplier;
+
+    return 0;
+}
+
+static long cmsx_applyTuningSliders(displayPort_t *pDisp, const void *self)
+{
+    UNUSED(pDisp);
+    UNUSED(self);
+
+    applyTuningSliders(currentPidProfile);
+
+    return 0;
+}
+
+static const OSD_Entry cmsx_menuSlidersEntries[] =
+{
+    { "-- PID SLIDERS --", OME_Label, NULL, NULL, 0},
+    { "PID SLIDERS",    OME_TAB,   NULL, &(OSD_TAB_t)   { &cmsx_slider_pids_mode, PID_TUNING_SLIDERS_MODE_COUNT - 1, lookupTableSliderPidsMode }, 0 },
+    { "MASTER MULT",    OME_FLOAT, NULL, &(OSD_FLOAT_t) { &cmsx_slider_master_multiplier, SLIDER_MIN, SLIDER_MAX, 1, 10 }, 0 },
+    { "R/P RATIO",      OME_FLOAT, NULL, &(OSD_FLOAT_t) { &cmsx_slider_roll_pitch_ratio,  SLIDER_MIN, SLIDER_MAX, 1, 10 }, 0 },
+    { "I GAIN",         OME_FLOAT, NULL, &(OSD_FLOAT_t) { &cmsx_slider_i_gain,            SLIDER_MIN, SLIDER_MAX, 1, 10 }, 0 },
+    { "PD RATIO",       OME_FLOAT, NULL, &(OSD_FLOAT_t) { &cmsx_slider_pd_ratio,          SLIDER_MIN, SLIDER_MAX, 1, 10 }, 0 },
+    { "PD GAIN",        OME_FLOAT, NULL, &(OSD_FLOAT_t) { &cmsx_slider_pd_gain,           SLIDER_MIN, SLIDER_MAX, 1, 10 }, 0 },
+    { "DMIN RATIO",     OME_FLOAT, NULL, &(OSD_FLOAT_t) { &cmsx_slider_dmin_ratio,        SLIDER_MIN, SLIDER_MAX, 1, 10 }, 0 },
+    { "FF GAIN",        OME_FLOAT, NULL, &(OSD_FLOAT_t) { &cmsx_slider_ff_gain,           SLIDER_MIN, SLIDER_MAX, 1, 10 }, 0 },
+
+    { "-- FILTER SLIDERS --", OME_Label, NULL, NULL, 0},
+    { "GYRO SLIDER",    OME_TAB,   NULL, &(OSD_TAB_t)   { &cmsx_slider_gyro_filter,  1, lookupTableOffOn }, 0 },
+    { "GYRO MULT",      OME_FLOAT, NULL, &(OSD_FLOAT_t) { &cmsx_slider_gyro_filter_multiplier,  SLIDER_MIN, SLIDER_MAX, 1, 10 }, 0 },
+    { "DTERM SLIDER",   OME_TAB,   NULL, &(OSD_TAB_t)   { &cmsx_slider_dterm_filter, 1, lookupTableOffOn }, 0 },
+    { "DTERM MULT",     OME_FLOAT, NULL, &(OSD_FLOAT_t) { &cmsx_slider_dterm_filter_multiplier, SLIDER_MIN, SLIDER_MAX, 1, 10 }, 0 },
+
+    { "APPLY SLIDERS",  OME_Funcall, cmsx_applyTuningSliders, NULL, 0 },
```

Comment:
We _could_ probably automate the application of the sliders in `cmsx_slidersOnExit()` if we detect that the user has changed any values - not sure what is the better UX in this case.

Reply:
The reason it is separate and not onExit is that onExit funtions are called when you save/save&exit... Meaning you couldn't apply sliders first then go and manually change pids because it would be recalculated on save. Detecting if users changed settings could work although I'm not sure how would I do that.

Reply:
Eh, good point. I guess you could override the 'back' menu entry with a custom function. And you could compare the shadow values that the parameters are copied into in `cmsx_slidersOnEnter()` to the actual parameters to detect changes.

Reply:
Sounds good. I'll try to do something like that.

Reply:
> We _could_ probably automate the application of the sliders in `cmsx_slidersOnExit()` if we detect that the user has changed any values - not sure what is the better UX in this case.

i've being trying it today the new simplified tuning and indeed, having to apply changes each time is a pain in the ass. auto-applying on exit would be be way better!

Reply:
I made #11001 to fix this.